tools/cdk-build-tools/lib/feature-flag.ts

```diff
@@ -0,0 +1,69 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as semver from 'semver';
+
+type Flags = {[key: string]: any} | undefined
+
+/* eslint-disable jest/no-export */
+
+/**
+ * jest helper function to be used when testing feature flags.
+ *
+ * This function is for writing jest tests that test feature flags that are present in CDKv1
+ * and will be removed in CDKv2. This helps execute tests written for CDKv1 in the context of CDKv2.
+ *
+ * The version of CDK is determined by running `scripts/resolve-version.js`, and the logic is as follows:
+ *
+ * When run in CDKv1, the tests are executed as normal.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is set
+ * (via CDK app context) are executed but without setting the app context. This is because CDKv2
+ * defaults to the behaviour for the flag being enabled.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is not
+ * set are skipped.
+ */
+export function withFeatureFlag<T>(
+  name: string,
+  flags: Flags,
+  cdkApp: new (flags: Flags) => T,
```

Comment:
This type isn't actually `flags: Flags`, btw, but `props: Props`. `cdk.App` takes

```ts
new cdk.App({
  context: { / * this is where Flags go! */ }, 
  outdir: { ... }, 
  //etc
})
```

Happens to work out because the types have the same shape, but the naming is wrong (and therefore slightly confusing).

If anything you'd want to write:

```ts
cdkApp: new (props?: { context?: Flags }) => T,
```